api/src/main/java/org/pmiops/workbench/db/jdbc/ReportingQueryServiceImpl.java

```diff
@@ -163,47 +174,81 @@ public long getQueryBatchSize() {
+                + "  GROUP_CONCAT(DISTINCT u.free_tier_credits_limit_days_override) AS free_tier_credits_limit_days_override,\n"
+                + "  GROUP_CONCAT(DISTINCT u.free_tier_credits_limit_dollars_override) AS free_tier_credits_limit_dollars_override,\n"
+                + "  GROUP_CONCAT(DISTINCT u.given_name) AS given_name,\n"
+                + "  GROUP_CONCAT(DISTINCT u.last_modified_time) AS last_modified_time,\n"
+                + "  GROUP_CONCAT(DISTINCT u.professional_url) AS professional_url,\n"
+                + "  GROUP_CONCAT(DISTINCT u.two_factor_auth_bypass_time) AS two_factor_auth_bypass_time,\n"
+                + "  GROUP_CONCAT(DISTINCT u.two_factor_auth_completion_time) AS two_factor_auth_completion_time,\n"
+                + "  GROUP_CONCAT(DISTINCT u.email) AS username,\n"
+                + "  GROUP_CONCAT(DISTINCT a.city) AS city,\n"
+                + "  GROUP_CONCAT(DISTINCT a.country) AS country,\n"
+                + "  GROUP_CONCAT(DISTINCT a.state) AS state,\n"
+                + "  GROUP_CONCAT(DISTINCT a.street_address_1) AS street_address_1,\n"
+                + "  GROUP_CONCAT(DISTINCT a.street_address_2) AS street_address_2,\n"
+                + "  GROUP_CONCAT(DISTINCT a.zip_code) AS zip_code,\n"
+                + "  GROUP_CONCAT(DISTINCT via.institution_id) AS institution_id,\n"
+                + "  GROUP_CONCAT(DISTINCT via.institutional_role_enum) AS institutional_role_enum,\n"
+                + "  GROUP_CONCAT(DISTINCT via.institutional_role_other_text) AS institutional_role_other_text,\n"
+                + "  GROUP_CONCAT(DISTINCT ud.degree) AS degrees,\n"
+                + "  GROUP_CONCAT(DISTINCT dm.ethnicity) AS ethnicity,\n"
+                + "  GROUP_CONCAT(DISTINCT dm.year_of_birth) AS year_of_birth,\n"
+                + "  GROUP_CONCAT(DISTINCT dm.disability) AS disability,\n"
+                + "  GROUP_CONCAT(DISTINCT dm.education) AS education,\n"
+                + "  GROUP_CONCAT(DISTINCT dm.identifies_as_lgbtq) AS identifies_as_lgbtq,\n"
+                + "  GROUP_CONCAT(DISTINCT dm.lgbtq_identity) AS lgbtq_identity,\n"
+                + "  GROUP_CONCAT(DISTINCT dm.gender_identity) AS gender_identity,\n"
+                + "  GROUP_CONCAT(DISTINCT dm.race) AS race,\n"
+                + "  GROUP_CONCAT(DISTINCT dm.sex_at_birth) AS sex_at_birth\n"
                 + "  \n"
                 + "FROM user u"
                 + "  LEFT OUTER JOIN address AS a ON u.user_id = a.user_id\n"
-                + "  LEFT OUTER JOIN user_verified_institutional_affiliation AS via on u.user_id = via.user_id"
+                + "  LEFT OUTER JOIN user_verified_institutional_affiliation AS via on u.user_id = via.user_id\n"
+                + "  LEFT OUTER JOIN user_degree AS ud on u.user_id = ud.user_id\n"
+                + "  LEFT OUTER JOIN "
+                + "  ( "
+                + "       SELECT \n"
```

Comment:
Is it not possible to push the GROUP_CONCAT + GROUP BY into this subquery? Likewise, could you do the same with a user_degree subquery?

I'm slightly worried about GROUP_CONCAT() on non-strings - it seems like the type rules are a bit weird here, check my logic: if singular: keep original type, else: coerce to comma-delimited string

Reply:
that is a good idea. I think it works.